compiler/parser/src/parser.rs

```diff
@@ -112,7 +112,7 @@ pub fn parse_tokens(
     let marker_token = (Default::default(), mode.to_marker(), Default::default());
     let tokenizer = iter::once(Ok(marker_token))
         .chain(lxr)
-        .filter_ok(|(_, tok, _)| !matches!(tok, Tok::Comment(_)));
+        .filter_ok(|(_, tok, _)| !matches!(tok, Tok::Comment { .. } | Tok::NonLogicalNewline));
```

Comment:
completely missed this too. could you just move the filtering into a different function instead? I doubt there's going to be many tokens that are filtered but its best if we keep it in a single function to avoid forgetting it in the future.

Reply:
Good idea; updated so that `parse_located` just calls `parse_tokens`, removing the entire duplication.